### PR TITLE
Add flip kernel and op

### PR DESCRIPTION
**Pull request by AlienLiang23**
Add flip kernel and op

### CODE REVIEW

**Comment by AlienLiang23** [comment]
image
image
passes all UTs

**Comment by fengyuan14** [comment]
Although we can see all dtpyes covered by non-fine-gran cases, we still need add the op into fine gran compare_cpu suite, since test_compare_cpu includes some PyTorch op semantics definition test.

File: src/aten/sycl/TensorTransformationsKernel.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <ATen/native/TensorIterator.h>
+
+namespace at::native::xpu {
+
+void flip_xpu_kernel(TensorIterator& iter);
```
Comment (fengyuan14) [inline src/aten/sycl/TensorTransformationsKernel.h:7]:
```suggestion
void flip_kernel(TensorIterator& iter);
```

File: src/aten/sycl/TensorTransformationsKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  const auto offset_calc =
+      make_offset_calculator<2, /*signed_strides=*/true>(iter);
+
+  FlipKernelImplLoopFunctor<scalar_t, decltype(offset_calc)> loop(
+      out_ptr, in_ptr, offset_calc);
+  launch_kernel(iter.numel(), loop);
+}
+
+void flip_xpu_kernel(TensorIterator& iter) {
```
Comment (fengyuan14) [inline src/aten/sycl/TensorTransformationsKernel.cpp:126]:
```suggestion
void flip_kernel(TensorIterator& iter) {
```

File: src/aten/TensorTransformations.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    // to do anything there
+    if (strides_dummy[i] == 0 && strides_self[i] != 0) {
+      data += strides_bytes[i] * (sizes[i] - 1);
+      strides_bytes[i] *= -1;
+    }
+  }
+  iter._unsafe_set_arg_strides(0, strides_bytes);
+  iter._unsafe_set_arg_data(0, reinterpret_cast<void*>(data));
+
+  at::native::xpu::flip_xpu_kernel(iter);
```
Comment (fengyuan14) [inline src/aten/TensorTransformations.cpp:77]:
```suggestion
  at::native::xpu::flip_kernel(iter);
```

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**